### PR TITLE
[REG master] Issue 15369 - id.d(369): Error: Outside Unicode code space

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=15369

### CODE REVIEW

File: src/backend/dt.c [hunk 1]
```diff
@@ -175,25 +175,25 @@ dt_t ** dtnbytes(dt_t **pdtend,unsigned size,const char *ptr)
...
-    dt->DTpbytes = (char *) MEM_PH_MALLOC(size + nbytes);
+    dt->DTnbytes = size + nzeros;
+    dt->DTpbytes = (char *) MEM_PH_MALLOC(size + nzeros);
     dt->Dty = ty;
     dt->DTabytes = offset;
     memcpy(dt->DTpbytes,ptr,size);
-    if (size)
-        memset(dt->DTpbytes + size, 0, nbytes);
+    if (nzeros)
+        memset(dt->DTpbytes + size, 0, nzeros);
```
Comment (9rnsr) [inline src/backend/dt.c]: I think it was merely a mistake.

**Comment by 9rnsr** [comment]
Ping: @WalterBright

File: src/backend/dt.c [hunk 2]
```diff
@@ -175,25 +175,25 @@ dt_t ** dtnbytes(dt_t **pdtend,unsigned size,const char *ptr)
  * Construct a DTabytes record, and return it.
  */
 
-dt_t **dtabytes(dt_t **pdtend, unsigned offset, unsigned size, const char *ptr, unsigned nbytes)
+dt_t **dtabytes(dt_t **pdtend, unsigned offset, unsigned size, const char *ptr, unsigned nzeros)
```
Comment (9rnsr) [inline src/backend/dt.c]: The name `nzeros` comes from `dt.h`.

File: src/todt.c [hunk 3]
```diff
@@ -388,6 +388,9 @@ dt_t **Expression_toDt(Expression *e, dt_t **pdt)
             {
                 case Tarray:
                     pdt = dtsize_t(pdt, n);
+                    pdt = dtabytes(pdt, 0, n * e->sz, p, (unsigned)e->sz);
```
Comment (yebblies) [inline src/todt.c]: This seems to be redundant.  Rest LGTM.

Comment (9rnsr) [inline src/todt.c]: When I'd like to insert a `printf` to `case Tpointer`, the case fallthrough was problematic.
In other words, the case fallthrough tangles the two cases.
I think the two lines redundancy is better so the two cases will be independent.

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Pull request merged**